rift/node.py

```diff
@@ -2804,7 +2804,10 @@ def spf_add_neighbor_candidates(self, node_system_id, node_cost, node_ties, cand
             if self.is_neighbor_bidirectional(node_system_id, nbr_system_id, nbr_tie_element,
                                               spf_direction, special_for_neg_disagg):
                 # We have found a feasible path to the neighbor node; is the best path?
-                cost = node_cost + nbr_tie_element.cost
+                if nbr_tie_element.cost >= common.constants.infinite_distance:
```

Comment:
The following code is not correct:
```
                if nbr_tie_element.cost >= common.constants.infinite_distance:
                    cost = common.constants.infinite_distance
                else:
                    cost = node_cost + nbr_tie_element.cost
```

Consider for example the case that node_cost = infinity - 1 and nbr_tie_element.cost = 5

The following code is correct (note that you don't need to worry about int32u overflow because Python integers have infinite range):

`cost = min(node_cost + nbr_tie_element.cost, common.constants.infinite_distance)
`